src/App.js

```diff
@@ -2,27 +2,50 @@ import React from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
-function useSemiPersistentState() {
-  
-  const [todoList, setTodoList] = React.useState(JSON.parse(localStorage.getItem('savedTodoList'))|| []
-  );
 
-  React.useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-  }, [todoList]);
+function App() {
 
-  return [todoList, setTodoList]
-}
+ //   create state for App with blank Array and 'todoList' variable and setter 'setTodoList'
 
+   
+ const [todoList, setTodoList] = React.useState([]);
+ const [isLoading, setIsLoading] = React.useState(true);
 
+    React.useEffect(
+      
+      ()=>{
+        
+        getAsyncTodoList().then((result)=>{
+          setTodoList(result.data.todoList);
+          setIsLoading(false);
+        });
+      
+      },[]);
 
-function App() {
+    
 
- //   create state for App with blank Array and 'todoList' variable and setter 'setTodoList'
+
+//--handler function--
+  const getAsyncTodoList = () => 
+
+  new Promise((resolve, reject) => 
+    setTimeout(
+      () => resolve({data: {todoList: JSON.parse(localStorage.getItem('savedTodoList')) } }), 
```

Comment:
you may want to do this`todoList: JSON.parse(localStorage.getItem('savedTodoList')) || []` so your code doesn't break

Reply:
fixed